test/utils/executor.hpp

```diff
@@ -54,60 +54,53 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #endif
 
 
-template <typename ExecType>
-std::shared_ptr<ExecType> init_executor(
-    std::shared_ptr<gko::ReferenceExecutor>);
-
-
-template <>
-inline std::shared_ptr<gko::ReferenceExecutor>
-    init_executor<gko::ReferenceExecutor>(
-        std::shared_ptr<gko::ReferenceExecutor>)
+inline void init_executor(std::shared_ptr<gko::ReferenceExecutor>,
+                          std::shared_ptr<gko::ReferenceExecutor>& exec)
```

Comment:
nit: I know it doesn't make a lot of difference, but the single parameter version would work as well in all of these (and the mpi) cases

Reply:
The template specialization version didn't work for me, because it can't differ in number of parameters afaik?

Reply:
Oh, I missed that. In that case, never mind.